### PR TITLE
Add full backup and restore options

### PR DESCRIPTION
**Pull request by ben-kaufman**
Add UI options for a user to fully backup the wallets and devices files and restore them from a backup.

### CODE REVIEW

File: src/cryptoadvance/specter/device_manager.py [hunk 1]
```diff
@@ -96,3 +100,15 @@ def remove_device(self, device, wallet_manager=None):
...
+    def devices_backup_file(self):
+        memory_file = BytesIO()
+        with zipfile.ZipFile(memory_file, 'w') as zf:
+            for device in self.devices.values():
+                data = zipfile.ZipInfo('{}.json'.format(device.name))
+                data.date_time = time.localtime(time.time())[:6]
+                data.compress_type = zipfile.ZIP_DEFLATED
+                zf.writestr('{}.json'.format(device.name), device.json)
+        memory_file.seek(0)
+        return memory_file
```
Comment (stepansnigirev) [inline src/cryptoadvance/specter/device_manager.py:114]: I am not sure it should be a property, better use a function.
When I am using property somewhere I assume that it's fast, so I can access it many times without carying much about caching.
But reading all files and zipping them may not be fast.

File: src/cryptoadvance/specter/specter.py [hunk 2]
```diff
@@ -314,3 +320,31 @@ def hwi_bridge_url(self):
...
+                for device in self.device_manager.devices.values():
+                    data = zipfile.ZipInfo('{}.json'.format(device.alias))
+                    data.date_time = time.localtime(time.time())[:6]
+                    data.compress_type = zipfile.ZIP_DEFLATED
+                    zf.writestr(
+                        'devices/{}.json'.format(device.alias),
+                        json.dumps(device.json)
+                    )
+        memory_file.seek(0)
+        return memory_file
```
Comment (stepansnigirev) [inline src/cryptoadvance/specter/specter.py:347]: Here as well, let's use a function instead of property.

**Comment by ben-kaufman** [comment]
@stepansnigirev I removed the property from `specter_backup_file` and completely deleted the `wallets_backup_file ` and `devices_backup_file` as they weren't used. Can you please check if now the performance is fine again?

**Comment by stepansnigirev** [comment]
Looks like we got a merge conflict as I merged #260. Could you rebase?

**Comment by ben-kaufman** [comment]
Resolved, should be fine now.

**Comment by stepansnigirev** [comment]
Still takes forever, not sure why. You don't have this issue, right?
I tried with python3.7 and 3.8, using bitcoin core 19.

**Comment by ben-kaufman** [comment]
No for me it works the same (python 3.8, core 20), that's very strange as I don't think there's a change here which could affect every call...

File: src/cryptoadvance/specter/wallet.py [hunk 3]
```diff
@@ -191,18 +201,24 @@ def from_json(cls, wallet_dict, device_manager, manager, default_alias='', defau
     def get_info(self):
         try:
             self.info = self.cli.getwalletinfo()
-        except:
+            if self.info['keypoolsize'] == 0:
+                self.keypoolrefill(0, end=self.keypool, change=False)
+                self.keypoolrefill(0, end=self.change_keypool, change=True)
```
Comment (stepansnigirev) [inline src/cryptoadvance/specter/wallet.py:206]: These three lines are making all requests increadibly long.
In Core v19 keypoolsize will be always zero.

Comment (ben-kaufman) [reply src/cryptoadvance/specter/wallet.py:206]: Hmmm so do you know if there's an alternative there valid for both 19 and 20?

**Pull request merged**